Clean up IAuthorizationRequirementData

See [Improved support for custom authorization policies with IAuthorizationRequirementData](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-4/#improved-support-for-custom-authorization-policies-with-iauthorizationrequirementdata)

ToDo:

- Fix all the null warnings and other warnings
- Fix 415 error

See the project [here](https://github.com/dotnet/AspNetCore.Docs.Samples/tree/main/security/authorization/AuthRequirementsData)

1. Create a token and DOB via [this command](https://github.com/dotnet/AspNetCore.Docs.Samples/blob/main/security/authorization/AuthRequirementsData/Authorization/MinimumAgePolicyProvider.cs#L55) and save token
2. run the app
3. `curl -i -H "Authorization: Bearer <token>" http://localhost:{port}/api/greetings/hello`

Output shows:

```
Evaluating authorization requirement for age >= 16
info: MinimumAgeAuthorizationHandler[0]
      Minimum age authorization requirement 16 satisfied
```

But request returns a 415 error.

@kevinchalet can you help? CC @jxwaters if you can help

contributes to https://github.com/dotnet/AspNetCore.Docs/issues/29291

cc @kevinchalet  @jxwaters 
Code now works by #149.

Just need to fix nullability warnings:

1>C:\GH\Auth\MinimumAgePolicyProvider.cs(40,38,40,52): warning CS8613: Nullability of reference types in return type of 'Task<AuthorizationPolicy> MinimumAgePolicyProvider.GetPolicyAsync(string policyName)' doesn't match implicitly implemented member 'Task<AuthorizationPolicy?> IAuthorizationPolicyProvider.GetPolicyAsync(string policyName)'.
1>C:\GH\Auth\MinimumAgePolicyProvider.cs(32,38,32,60): warning CS8613: Nullability of reference types in return type of 'Task<AuthorizationPolicy> MinimumAgePolicyProvider.GetFallbackPolicyAsync()' doesn't match implicitly implemented member 'Task<AuthorizationPolicy?> IAuthorizationPolicyProvider.GetFallbackPolicyAsync()'.
1>C:\GH\Auth\MinimumAgePolicyProvider.cs(33,29,33,76): warning CS8619: Nullability of reference types in value of type 'Task<AuthorizationPolicy?>' doesn't match target type 'Task<AuthorizationPolicy>'.
1>C:\GH\Auth\MinimumAgePolicyProvider.cs(51,53,51,57): warning CS8625: Cannot convert null literal to non-nullable reference type.

Update, only one warning left of [this line](https://github.com/dotnet/AspNetCore.Docs.Samples/blob/main/security/authorization/AuthRequirementsData/Authorization/MinimumAgePolicyProvider.cs#L48)

warning CS8619: Nullability of reference types in value of type 'Task<AuthorizationPolicy>' doesn't match target type 'Task<AuthorizationPolicy?>'.

--- a/security/authorization/AuthRequirementsData/Authorization/MinimumAgePolicyProvider.cs
+++ b/security/authorization/AuthRequirementsData/Authorization/MinimumAgePolicyProvider.cs
@@ -43,9 +43,9 @@ class MinimumAgePolicyProvider : IAuthorizationPolicyProvider
             int.TryParse(policyName.Substring(POLICY_PREFIX.Length), out var age))
         {
             var policy = new AuthorizationPolicyBuilder(
-                                                 JwtBearerDefaults.AuthenticationScheme);
+                                                JwtBearerDefaults.AuthenticationScheme);
             policy.AddRequirements(new MinimumAgeRequirement(age));
-            return Task.FromResult(policy.Build());
+            return Task.FromResult<AuthorizationPolicy?>(policy.Build());
         }
 
         return Task.FromResult<AuthorizationPolicy?>(null);

